analysis/portanalyzer.go

```diff
@@ -0,0 +1,165 @@
+package analysis
+
+import (
+	"log"
+	"math"
+
+	"github.com/sarchlab/akita/v3/sim"
+)
+
+// PortBandwidthLogger is a hook for logging messages as they go across a Port
+type PortBandwidthLogger struct {
+	sim.LogHookBase
+	sim.TimeTeller
+
+	name      string
+	startTime sim.VTimeInSec
```

Comment:
I renamed most of the fields to lower case. There is no point in exporting them. We should encapsulate as much as possible.